modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerImpl.java

```diff
@@ -976,7 +987,18 @@ public void repopulate() throws IndexRebuildException {
 
                 var updatedEventData = index.getEvent(snapshot.getMediaPackage().getIdentifier().toString(), orgId,
                     snapshotSystemUser);
-                updatedEventData = getEventUpdateFunction(snapshot, orgId, snapshotSystemUser).apply(updatedEventData);
+                if (ServicePart.ACL.equals(parsedType)) {
```

Comment:
Reverse the order of checks here to adapt to the suggestion about.  If `type` is `null`, we don't need to worry about `parsedType` at all anymore.  If it's not then `parsedType` won't be `null`, at the very least.